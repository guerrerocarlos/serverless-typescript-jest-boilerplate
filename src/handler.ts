"use strict";
import lambda from "aws-lambda";

function lambdaResponse(statusCode: number, body: any) {
  return { statusCode, body: JSON.stringify(body) };
}

export async function hello(event: lambda.APIGatewayEvent) {
  return lambdaResponse(200, { result: "OK", event });
}
