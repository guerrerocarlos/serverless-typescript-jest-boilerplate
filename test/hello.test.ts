import { hello as helloHandler } from "../src/handler";
import lambda from "aws-lambda";

describe("Test /hello", () => {
  test("Basic handler working", async () => {
    let event = {};
    let response = helloHandler(event as lambda.APIGatewayEvent);

    expect(response).toMatchSnapshot();
  });
});
