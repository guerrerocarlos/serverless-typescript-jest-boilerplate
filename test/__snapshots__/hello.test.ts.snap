// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test /hello Normal payment flow 1`] = `Promise {}`;
